tools/rtpsrelay/GuidAddrSet.cpp

```diff
@@ -64,29 +67,32 @@ GuidAddrSet::record_activity(const Proxy& proxy,
   relay_participant_status_reporter_.set_alive_active(proxy, src_guid, true, true);
 
   {
-    AddrSet& addr_set = *addr_set_stats.select_addr_set(remote_address.port);
-    const auto res = addr_set.insert(std::make_pair(remote_address, expiration));
-    if (res.second) {
-      if (config_.log_activity()) {
-        ACE_DEBUG((LM_INFO, "(%P|%t) INFO: GuidAddrSet::record_activity "
-                   "%C %C is at %C %C into session addrs=%B total=%B remote=%B deactivation=%B expire=%B admit=%B\n",
-                   handler.name().c_str(),
-                   guid_to_string(src_guid).c_str(),
-                   OpenDDS::DCPS::LogAddr(remote_address.addr).c_str(),
-                   addr_set_stats.get_session_time(now).sec_str().c_str(),
-                   addr_set.size(),
-                   guid_addr_set_map_.size(),
-                   remote_map_.size(),
-                   deactivation_guid_queue_.size(),
-                   expiration_guid_addr_queue_.size(),
-                   admission_control_queue_.size()));
+    AddrSet& addr_set = *addr_set_stats.select_addr_set(remote_address.port, now);
```

Comment:
Why not just wrap the existing code with `if (config_.max_addr_set_size() == 0 || addr_set.size() <= config_.max_addr_set_size()) {`?

Reply:
You don't want to just wrap the existing code because then when the `addr_set` size is equal to the max, the expiration updates won't get processed. So you have to check to see if it would be increasing the size first, and then limit the increases but not the updates. However, revisiting this code does highlight the fact that we don't technically need the external `if` block, so we can simplify that part.